src/easy/handler.rs

```diff
@@ -379,8 +379,12 @@ pub struct Easy2<H> {
     inner: Box<Inner<H>>,
 }
 
+/// A thin wrapper around a low-level Easy handle that performs
+/// cleanup when dropped.
+struct CURLPtr(*mut curl_sys::CURL);
```

Comment:
This is necessary because `Inner` can't be `Drop` anymore due to [E0509](https://doc.rust-lang.org/error_codes/E0509.html), and because the following workarounds are unavailable:
- We can't use `std::mem::take` because H isn't `Default`.
- We don't want to use an `Option` because it adds overhead.